addons/website_sale_stock/models/sale_order.py

```diff
@@ -0,0 +1,22 @@
+# -*- coding: utf-8 -*-
+from openerp import SUPERUSER_ID
+from openerp.osv import osv
+from openerp.tools.translate import _
+
+class sale_order(osv.Model):
+    _inherit = "sale.order"
+
+    def _cart_update(self, cr, uid, ids, product_id=None, line_id=None, add_qty=0, set_qty=0, context=None, **kwargs):
+        """
+            display a out of stock warning message on cart, if customize option `Out of Stock Warning`
+            is set active and product type is not `Consumable` or `Service`
+         """
+        values = {}
+        customize_option = self.pool['ir.model.data'].xmlid_to_object(cr, SUPERUSER_ID, 'website_sale_stock.products_out_of_stock_warning')
```

Comment:
yes , to check customize option is active or not , if active we will show a warning message.